dts/arm/nordic/nrf5340_cpuapp_ipc.dtsi

```diff
@@ -5,12 +5,14 @@
  */
 
 ipc0: ipc0 {
-	compatible = "zephyr,ipc-openamp-static-vrings";
-	memory-region = <&sram0_shared>;
+	compatible = "zephyr,ipc-icbmsg";
+	status = "okay";
 	mboxes = <&mbox 0>, <&mbox 1>;
 	mbox-names = "tx", "rx";
-	role = "host";
-	status = "okay";
+	tx-region = <&cpuapp_cpunet_ipc_shm>;
+	rx-region = <&cpunet_cpuapp_ipc_shm>;
+	tx-blocks = <32>;
+	rx-blocks = <32>;
```

Comment:
Since we have 32K for each direction, we will get almost 1K per block. We can increase the number of blocks, because it has almost no cost and it will give better granularity. I'm suggesting 64 blocks.

Reply:
Currently it's a cherry-pick from upstream, so I would suggest limited changes here, and changing in the upstream. The openamp worked with 32 blocks as I remember and there were no issues